cluster/gce/util.sh

```diff
@@ -2852,11 +2852,26 @@ function setup-proxy() {
   ssh-to-node ${PROXY_NAME} "sudo apt update -y"
   ssh-to-node ${PROXY_NAME} "sudo apt install -y ${KUBERNETES_SCALEOUT_PROXY_APP}"
 
+  patch-haproxy-prometheus
+
   ssh-to-node ${PROXY_NAME} "sudo sed -i '/^ExecStart=.*/a ExecStartPost=/bin/bash -c \"sleep 20 && for npid in \$(pidof ${KUBERNETES_SCALEOUT_PROXY_APP}); do sudo prlimit --pid \$npid --nofile=500000:500000 ; done\"' /lib/systemd/system/${KUBERNETES_SCALEOUT_PROXY_APP}.service"
   ssh-to-node ${PROXY_NAME} "sudo systemctl daemon-reload"
   ssh-to-node ${PROXY_NAME} "sudo systemctl restart ${KUBERNETES_SCALEOUT_PROXY_APP}"
 }
 
+function patch-haproxy-prometheus {
+  # based on https://www.haproxy.com/blog/haproxy-exposes-a-prometheus-metrics-endpoint
+  ssh-to-node ${PROXY_NAME} "echo 'Patching Haproxy to expose prometheus...'"
```

Comment:
Minor: Run "echo 'Patching Haproxy to expose prometheus...'" directly on the test machine instead of the proxy machine?

Reply:
fixed